### ISSUE TITLE
Restore support for the NEAR search operator (from 711)

### ISSUE THREAD

**Issue by gigamorph** [project-lux/lux-marklogic#20]
**Problem Description:**
NEAR was excluded from [#653 - Switch from ML Search grammer to JSON-formatted search criteria](https://git.yale.edu/lux-its/marklogic/issues/653)'s scope, but is desired. 

Note: Lower priority than BOOST ([#710](https://git.yale.edu/lux-its/marklogic/issues/710)) | FYI: Boost has already been implemented in 2022

**Expected Behavior/Solution:**
Implement the NEAR search operator in LUX so users can find records with certain words NEAR each other within a document. 

**Requirements:**
- NEAR search operator will be implemented and supported in LUX. 
- Do not wrap cts.tripleRangeQuery in cts.nearQuery.

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] ~Wireframe/Mockup - Mike~
- [ ] Update FAQ text - Sarah/Amy
- [ ] Committee discussions - Sarah
- [x] Feasibility/Team discussion - Sarah
- [x] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] ~Are new regression tests required for QA - Amy~ (unsure) 
- [ ] Questions
- Are there any test cases for this? 

**UAT/LUX Examples:**

- https://lux.collections.yale.edu/view/results/works?q=%7B%22AND%22%3A%5B%7B%22text%22%3A%22Shakespeare%22%2C%22_lang%22%3A%22en%22%7D%2C%7B%22text%22%3A%22near%22%2C%22_lang%22%3A%22en%22%7D%2C%7B%22text%22%3A%22hamlet%22%2C%22_lang%22%3A%22en%22%7D%5D%7D&sq=Shakespeare+NEAR+hamlet (will return results of all documents which have the word shakespeare within five words away from the word hamlet) 

**Dependencies/Blocks:**

- Blocked By: Issues that are blocking the completion of the current issue.
- Blocking: Issues being blocked by the completion of the current issue.

**Related Github Issues:**

- Issues that contain similar work but are not blocking or being blocked by the current issue.

**Related links:**

- These links can consist of resources, bugherds, etc.

**~Wireframe/Mockup:~**
~Place wireframe/mockup for the proposed solution at end of ticket.~

**Comment by brent-hartwig** [comment]
@roamye, @prowns and @azaroth42, similar question as #21: do we anticipate any LUX backend endpoint consumer to want this feature?  The perplexing bit was defining "near" if one word was in the search result document and another was in the related document.  We need not support that.  We could require both words within the search result document _or_ a related document.  There's not as much code to remove as #21 but we could close the ticket.

**Comment by azaroth42** [comment]
Agreed on perplexity, and that requiring the words within the same document is fine. I would suggest finish forming the ticket and throw into backlog, priority low

**Comment by brent-hartwig** [comment]
@roamye, do you have a sense of priority?  I ask because we have quite a few tickets in Forming.

**Comment by roamye** [comment]
@brent-hartwig - this has been formatted and put into prioritization review.

**Comment by brent-hartwig** [comment]
@azaroth42 and @clarkepeterf, I just ran across the [triple positions indexing option](https://docs.marklogic.com/guide/semantics/indexes#id_34194) and the four contexts where it improves search result accuracy or, in some cases, makes the positional search possible.

1. cts.nearQuery: I'm wondering if the example returns documents containing triples that have the specified subjects within 11 triples of each other.  If true and given the pipeline controls the order of the triples, _maybe_ some value could be derived --but not in the keyword search context we have been thinking of.  Interesting but valuable?
2. cts.elementQuery: This is another proximity bit but appears to be triples within XML elements or JSON properties.  Here too, if we could "create knowledge" by placing triples within additional structure, we could utilize in our queries.
3. The item-frequency option of cts.triples: This one makes me think of the cardinality search discussed in #49, but strictly for triples.
4. cts.tripleRangeQuery: This one I don't have a theory on.  The word "position" is not in its API documentation.

Without looking into it, I don't know how these would translate to Optic.

**Comment by prowns** [comment]
@roamye - this will also need a front-end ticket (Advanced Search, help text, etc.)

**Comment by roamye** [comment]
Moving this to blocked as we need to define FE requirements.
